src/Connection/Cluster/RedisCluster.php

```diff
@@ -337,10 +337,19 @@ protected function createConnection($connectionID)
     {
         $separator = strrpos($connectionID, ':');
 
-        return $this->connections->create([
+        $parameters = [
             'host' => substr($connectionID, 0, $separator),
             'port' => substr($connectionID, $separator + 1),
-        ]);
+        ];
+
+        $existConnection = current($this->pool);
+        if ($existConnection instanceof NodeConnectionInterface) {
+            $existParameters = $existConnection->getParameters()->toArray();
+            unset($existParameters['alias'], $existParameters['slots']);
```

Comment:
@tillkruss Removed slots as well, shouldn't be applied to different connection if already covered by another one